tests/ci_tests/fixtures/prepare_environment_fixture.py

```diff
@@ -50,40 +55,154 @@ def _setup_database(db_conn):
         __insert_into_tables(db_conn, model_setup)
 
 
-def _create_aws_connection(conn):
+@pytest.fixture(scope="session")
+def connection_object_for_aws_credentials(db_conn, aws_s3_bucket):
+    aws_conn_name = "test_aws_credentials_connection_name"
+    aws_region = os.environ["AWS_DEFAULT_REGION"]
+    aws_s3_uri = f"https://{aws_s3_bucket}.s3.{aws_region}.amazonaws.com"
     query = "CREATE OR REPLACE  CONNECTION {aws_conn_name} " \
             "TO '{aws_s3_uri}' " \
-            "USER '{aws_key_id}' IDENTIFIED BY '{aws_access_key}'"\
-        .format(aws_conn_name=aws_params.aws_conn_name,
-                aws_s3_uri=aws_params.aws_s3_uri,
-                aws_key_id=os.environ["AWS_ACCESS_KEY_ID"],
-                aws_access_key=os.environ["AWS_SECRET_ACCESS_KEY"])
-    conn.execute(query)
+            "USER '{aws_access_key_id}' IDENTIFIED BY '{aws_secret_access_key}'" \
+        .format(aws_conn_name=aws_conn_name,
+                aws_access_key_id=os.environ["AWS_ACCESS_KEY_ID"],
+                aws_secret_access_key=os.environ["AWS_SECRET_ACCESS_KEY"],
+                aws_s3_uri=aws_s3_uri)
+    db_conn.execute(query)
+    yield aws_conn_name
+    db_conn.execute(f"DROP CONNECTION {aws_conn_name};")
 
 
 def _create_aws_s3_bucket():
     s3_client = boto3.client('s3')
+    bucket_name = "ci-exasol-sagemaker-extension-bucket"
     try:
         s3_client.create_bucket(
-            Bucket=aws_params.aws_bucket,
+            Bucket=bucket_name,
             CreateBucketConfiguration={
                 'LocationConstraint': os.environ["AWS_DEFAULT_REGION"]}
         )
     except s3_client.exceptions.BucketAlreadyOwnedByYou as ex:
         print("'BucketAlreadyOwnedByYou' exception is handled")
+    return bucket_name
 
 
-def _remove_aws_s3_bucket():
+def _remove_aws_s3_bucket_content(bucket_name: str):
     s3_client = boto3.resource('s3')
-
-    bucket = s3_client.Bucket(aws_params.aws_bucket)
+    bucket = s3_client.Bucket(bucket_name)
     bucket.objects.all().delete()
 
 
 @pytest.fixture(scope="session")
-def prepare_ci_test_environment(db_conn):
+def aws_s3_bucket():
+    bucket_name = _create_aws_s3_bucket()
+    yield bucket_name
+    _remove_aws_s3_bucket_content(bucket_name)
+
+
+@pytest.fixture(scope="session")
+def aws_sagemaker_role() -> str:
+    iam_client = boto3.client('iam')
+    role_name = _create_sagemaker_role(iam_client)
+    policy_arn = _create_sagemaker_policy(iam_client)
+    _attach_policy_to_role(iam_client,
+                           policy_arn=policy_arn,
+                           role_name=role_name)
+    _attach_policy_to_role(iam_client,
+                           policy_arn="arn:aws:iam::aws:policy/AmazonSageMakerFullAccess",
+                           role_name=role_name)
+    return role_name
```

Comment:
Do we need to delete the role afterwards, like the bucket in `aws_s3_bucket()`?
Same for the following functions.

Reply:
The actual bucket doesn't get deleted, only the content gets deleted. I don't think, we need to delete the role, it can be reused in a future run and does get deleted anyway every night. From my knowledge, it doesn't really produce costs.